wasm/test_page/js/worker.js

```diff
@@ -78,7 +78,7 @@ onmessage = async function(e) {
 // Instantiates the Translation Service
 const constructTranslationService = async () => {
   if (!translationService) {
-    var translationServiceConfig = {};
+    var translationServiceConfig = {cacheEnabled: true, cacheSize: 20000};
```

Comment:
Sentences are small.  The problem is all the other stuff like alignment.

Reply:
`History` is a bigger object than a sentence (sometimes contains alignment matrix which is order N^2 to source-target token lengths).  The intended message is you have control over how many History items you want to cache. The real-world concept we want to map to is going to be how many sentences repeat across UI elements in pages under the same subdomain (Imagine you're Wikipedia link surfing and repeatedly translating the sidebar, same with a news website). I expect this to be useful at @jelmervdl's extension with a shared state among tabs. For the Mozilla extension, this might just be useful for outbound translation you can assume people don't enter larger than some `n` sentences and keep it small. 

Say, there's going to be about 12 tokens in 50 bytes. Assuming the same length in both languages, 12 x 12 tokens lead to a 144 size matrix. Each of those floats are 4 bytes, so 144 x 4. We also store a few more information (unnormalized score etc) so it's going to be a bit more - but you get the picture. translateLocally appears to have gone with a default of `1<<16` (https://github.com/XapaJIaMnu/translateLocally/pull/75).

Since the container is just storing `Ptr<History>` there are only `sizeof(std::shared_ptr<History>) * cacheSize`  to exist. As storage increases, this will grow. Hitting full occupancy with a random-eviction policy is tricky, one may not end up using the full capacity. We can iterate to add more code to support more complicated storage-size based limits if there are tight memory constraints that show up, but first I'd want to see this in action and if it's especially useful.

Reply:
So, each history item's ram requirement is O(N*M) where N, M are the no. of tokens in source and target sentence respectively.